### PR TITLE
Client option no_proxy does not work as advertised

### PR DESCRIPTION
**Pull request by chefsalim**
Update no_proxy usage in chef-config to conform to documentation:
https://docs.chef.io/proxies.html#no-proxy
http://www.gnu.org/software/emacs/manual/html_node/url/Proxies.html

@chef/client-core @randomcamel @tyler-ball

### CODE REVIEW

File: chef-config/chef-config.gemspec [hunk 1]
```diff
@@ -17,6 +17,7 @@ Gem::Specification.new do |spec|
 
   spec.add_dependency "mixlib-shellout", "~> 2.0"
   spec.add_dependency "mixlib-config", "~> 2.0"
+  spec.add_dependency "fuzzyurl", '~> 0.8.0'
```
Comment (tyler-ball) [inline chef-config/chef-config.gemspec]: This is released under the MIT license, is that okay for our usage?

**Comment by tyler-ball** [comment]
:+1:

File: chef.gemspec [hunk 2]
```diff
@@ -31,6 +31,7 @@ Gem::Specification.new do |s|
   s.add_dependency "diff-lcs", "~> 1.2", ">= 1.2.4"
 
   s.add_dependency "chef-zero", "~> 4.5"
+  s.add_dependency "fuzzyurl", '~> 0.8.0'
```
Comment (randomcamel) [inline chef.gemspec]: Is this necessary, since `chef` requires `chef-config` which already declares it, and it's really `chef-config` that uses it?

**Comment by randomcamel** [comment]
One question, otherwise :+1:

File: chef-config/chef-config.gemspec (hunk 1 above)
Comment (chefsalim) [inline chef-config/chef-config.gemspec]: Yes, confirmed.

**Comment by tyler-ball** [comment]
Nice!

**Comment by sergiopatino** [comment]
Any idea what version of chef-client this made it into ?

**Comment by tyler-ball** [comment]
@sergiopatino It made it into `12.9.38` - I'm not sure why it isn't in the CHANGELOG.md though...

**Pull request merged**